src/execution/values.js

```diff
@@ -64,9 +67,12 @@ export function getVariableValues(
         ),
       );
     } else {
+      const hasValue = hasOwnProperty(inputs, varName);
       const value = inputs[varName];
-      if (isInvalid(value)) {
+      if (!hasValue || isNullish(value)) {
```

Comment:
`isNullish` checks that `value !== value` that is not a problem if variables always comes from JSON. But I see a potential problem with NaN if variables come from some binary protocol or created by JS.

BTW. Nothing prevents a user from creating custom scalar like `DoubleIEEE754` that accepting `NaN`, `Infinity`, etc.

Reply:
That's fair. This could be a more general `== null` check